netlink-sys/src/sys.rs

```diff
@@ -263,6 +263,18 @@ impl Socket {
         Ok(res as usize)
     }
 
+    pub fn recv_all(&self) -> Result<(Vec<u8>, SocketAddr)> {
```

Comment:
If I understand correctly, unlike `recv_from` which can truncate the message, `recv_all` will return the full message, correct? It would be nice to document that behavior by adding a docstring to both `recv_from` and `recv_all`.


Also what do you think about renaming `recv_all` into something like `recv_from_full` or `recv_from_no_truncate`? It doesn't looks as nice as `recv_all` but it makes it clearer that this is basically a variant of `recv_from`, imo.

Reply:
@little-dude  I think these two names are not good names, even `recv_from` is. 
`recv_from` from where/who?  the address pointer is not exposed, right? 
I think these names should not rely on underlying libc function name, we just name it by its action. maybe one day we could replace all libc::recvfrom to libc::recvmsg without rename the function

How about rename `recv_all` to `recv_all_available` avoid confusion with `recv_to_end`?

Reply:
`recv_from` comes from the `recvfrom` C API. Sure we don't pass a function pointer as an argument, but we do return the sender's address.

```
ssize_t recv(int sockfd, void *buf, size_t len, int flags);
```

becomes 

```
fn recv(&self, buf: &mut [u8], flags: libc::c_int) -> Result<usize>
``` 

and 

```
ssize_t recvfrom(int sockfd, void *buf, size_t len, int flags,
                 struct sockaddr *src_addr, socklen_t *addrlen);
```

becomes

```
pub fn recv_from(&self, buf: &mut [u8], flags: libc::c_int) -> Result<(usize, SocketAddr)>
```

The function that you introduced is basically a form of `recvfrom` so that's why I'd like to keep that naming.

Reply:
well, I will rename it to recv_from_full just hope to merge quickly.